crates/bevy_ecs/src/query/default_filters.rs

```diff
@@ -0,0 +1,167 @@
+use crate as bevy_ecs;
+use crate::{component::ComponentId, query::FilteredAccess};
+use bevy_ecs_macros::Resource;
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+enum FilterKind {
+    With,
+    Without,
+}
+
+#[derive(PartialEq, Eq, Debug)]
+struct DefaultFilter {
+    component_id: ComponentId,
+    kind: FilterKind,
+}
+
+/// A list of default query filters, these can be used to globally exclude entities from queries.
+/// Each individual filter is only applied to queries that don't mention that component.
+///
+/// These filters are only applied to queries initialized after updating this resource,
+/// it should most likely only be modified before the app starts.
+#[derive(Resource, Default)]
+pub struct DefaultQueryFilters(Vec<DefaultFilter>);
+
+impl DefaultQueryFilters {
+    /// Add a With filter to the default query filters
+    pub fn with(&mut self, component_id: ComponentId) {
+        self.set(component_id, FilterKind::With);
+    }
+
+    /// Add a Without filter to the default query filters
+    pub fn without(&mut self, component_id: ComponentId) {
+        self.set(component_id, FilterKind::Without);
+    }
+
+    /// Remove a filter for the specified [`ComponentId`]
+    pub fn remove(&mut self, component_id: ComponentId) {
+        self.0.retain(|filter| filter.component_id != component_id);
+    }
+
+    fn set(&mut self, component_id: ComponentId, kind: FilterKind) {
+        if let Some(filter) = self
+            .0
+            .iter_mut()
+            .find(|filter| filter.component_id == component_id)
+        {
+            filter.kind = kind;
+        } else {
+            self.0.push(DefaultFilter { component_id, kind });
+        }
+    }
+
+    pub(super) fn apply(&self, component_access: &mut FilteredAccess<ComponentId>) {
+        for filter in self.0.iter() {
+            if !contains_component(component_access, filter.component_id) {
+                match filter.kind {
+                    FilterKind::With => component_access.and_with(filter.component_id),
+                    FilterKind::Without => component_access.and_without(filter.component_id),
+                }
+            }
+        }
+    }
+}
+
+fn contains_component(
```

Comment:
The way it functions might not be the expected behavior of future logic needing something like it. But if other code needs the same check later it can of course be moved there.